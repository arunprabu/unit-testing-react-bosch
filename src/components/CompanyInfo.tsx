const CompanyInfo = (props: any) => {
  return (
    <div>
      <h2>Company Info!!....</h2>
      <p>
        Company Name:
        <span>{props.companyName}</span>
      </p>

      <p>
        Founder Year:
        <span>{props.foundedYear}</span>
      </p>

      <p>
        Employees Count:
        <span>{props.employeesCount}</span>
      </p>
    </div>
  );
};

export default CompanyInfo;
