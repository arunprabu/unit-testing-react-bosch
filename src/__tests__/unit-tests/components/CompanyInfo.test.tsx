// AAA (Arrange, Act, Assert )

import { render, screen } from "@testing-library/react"; // render the sample
import React from "react";
import CompanyInfo from "../../../components/CompanyInfo"; // arraging the sample

// Test suite = group of related test specs
describe("CompanyInfo", () => {
  // test case #1 a.k.a test, test spec
  it("has heading 'Company Info'", () => {
    render(<CompanyInfo />);

    const heading = screen.getByText(/Company info/i);
    expect(heading).toBeTruthy();
  });

  // test case #2
  // Prop Testing
  it("receives props properly and displays them", () => {
    render(
      <CompanyInfo
        companyName="IBM"
        foundedYear="1950"
        employeesCount="23457"
      />
    );
    const companyElement = screen.getByText("IBM");
    expect(companyElement).toBeDefined();

    const foundedYearElement = screen.getByText("1950");
    expect(foundedYearElement).toBeDefined();

    const employeesCountElement = screen.getByText("23457");
    expect(employeesCountElement).toBeDefined();
  });

  // Snapshot Testing -- old one
  // it("renders the snapshot correctly", () => {
  //   const tree = renderer.create(
  //     <CompanyInfo
  //       companyName="IBM"
  //       foundedYear="1950"
  //       employeesCount="23457"
  //     />
  //   ).toJSON();

  //   expect(tree).toMatchSnapshot();
  // });

  it("renders the snapshot correctly -- modern approach", () => {
    const { asFragment } = render(
      <CompanyInfo
        companyName="IBM"
        foundedYear="1950"
        employeesCount="23457"
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
