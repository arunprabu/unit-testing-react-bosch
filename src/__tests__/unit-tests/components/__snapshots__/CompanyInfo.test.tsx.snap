// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyInfo renders the snapshot correctly -- modern approach 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Company Info!!....
    </h2>
    <p>
      Company Name:
      <span>
        IBM
      </span>
    </p>
    <p>
      Founder Year:
      <span>
        1950
      </span>
    </p>
    <p>
      Employees Count:
      <span>
        23457
      </span>
    </p>
  </div>
</DocumentFragment>
`;
